src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -1752,11 +1789,37 @@ ValidateClauseList(List *clauseList)
 	{
 		Node *clause = (Node *) lfirst(clauseCell);
 
-		bool selectClause = IsSelectClause(clause);
-		bool joinClause = IsJoinClause(clause);
-		bool orClause = or_clause(clause);
+		/* produce a better error message for sublinks */
```

Comment:
No that's not. Below we error out only saying `unsupported clause type`. But for sublinks we can produce better message